File: LeetCodeTasks/3019_number-of-changing-keys/solutions/java/solution.java
class Solution {
    public int countKeyChanges(String s) {
        int changes = 0;
        char prev = '\0'; // Initialize prev to a non-letter character
        for (char c : s.toCharArray()) {
            if (Character.isLetter(c) && Character.toLowerCase(c) != prev) {
                changes++;
                prev = Character.toLowerCase(c);
            }
        }
        return changes;
    }
}

File: LeetCodeTasks/3019_number-of-changing-keys/solutions/c/solution.c
#include <stdbool.h>

int countKeyChanges(char* s) {
    int changes = 0;
    bool isUpperCase = false;

    for (int i = 0; s[i] != '\0'; i++) {
        if (s[i] >= 'A' && s[i] <= 'Z') {
            if (!isUpperCase) {
                isUpperCase = true;
                changes++;
            }
        } else {
            if (isUpperCase) {
                isUpperCase = false;
                changes++;
            }
        }
    }

    return changes;
}
